iXBRLViewerPlugin/viewer/src/js/index.js

```diff
@@ -55,24 +55,40 @@ $(function () {
     var inspector = new Inspector();
     setTimeout(function(){
 
-        var iframe = reparentDocument();
+        var iframes = $(reparentDocument());
+
+        var taxonomyData = getTaxonomyData();
+        if (taxonomyData === null) {
+            $('#ixv .loader .text').text("Error: Could not find viewer data");
+            $('#ixv .loader').removeClass("loading");
+            return;
+        }
+        var report = new iXBRLReport(JSON.parse(taxonomyData));
+        if (report.isDocumentSet()) {
+            $('#ixv .ixds-tabs').show();
+            var ds = report.documentSetFiles();
+            for (var i = 1; i < ds.length; i++) {
+                var iframe = $("<iframe src=\"" + ds[i] + "\" />").appendTo("#ixv #iframe-container");
+                iframes = iframes.add(iframe);
+            }
+        }
 
         /* Poll for iframe load completing - there doesn't seem to be a reliable event that we can use */
```

Comment:
Out of curiosity, why doesn't `$(...).ready(...)` work?

Reply:
I'm not entirely sure, but I've just tried it and it doesn't seem to.  I remember searching when I originally wrote this and confirming that there wasn't an easy way.  I'm not sure if it was specific to the fact that we dynamically populate the iframe.